spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.spec.js

```diff
@@ -149,24 +149,47 @@ describe('GA4 core', function () {
       })
     })
 
-    it('correctly identifies a relative URL', function () {
-      var href = '/relativeURL'
-      expect(GOVUK.analyticsGa4.core.trackFunctions.hrefIsRelative(href)).toEqual(true)
+    it('correctly identifies relative URLs', function () {
+      var relativeURLs = [
+        'g',
+        './g',
+        'g/',
+        '/g',
+        '?y',
+        'g?y',
+        'g?y/./x',
+        '#s',
+        'g#s',
+        'g#s/./x',
+        'g?y#s',
+        ';x',
+        'g;x',
+        'g;x?y#s',
+        '.',
+        './',
+        '..',
+        '../',
+        '../g',
+        '../..',
+        '../../',
+        '../../g']
+
+      for (var i = 0; i < relativeURLs.length; i++) {
+        var href = relativeURLs[i]
+        expect(GOVUK.analyticsGa4.core.trackFunctions.hrefIsRelative(href)).toEqual(true)
```

Comment:
Can this also test that the function doesn't error when passed nothing?

Reply:
Thanks @andysellick, I've added a check for an empty string, and nothing being passed :+1: